src/components/customers/CustomerPaymentMethods.tsx

```diff
@@ -46,48 +53,49 @@ gql`
 
 export const CustomerPaymentMethods = ({ customer, linkedPaymentProvider }: Props) => {
   const { translate } = useInternationalization()
+  const addPaymentDialogRef = useRef<DialogRef>(null)
+  const [selectedPaymentProvider, setSelectedPaymentProvider] = useState<string>('')
+
   const [generateCheckoutUrlMutation, { data, loading, error, reset }] =
     useGenerateCheckoutUrlMutation({
       variables: {
         input: { customerId: customer.id },
       },
     })
 
-  const addPaymentDialogRef = useRef<DialogRef>(null)
+  const hasOnlyIneligiblePaymentMethods = useMemo(() => {
+    const linkedProviderCustomer = customer.providerCustomer
+    const availableProviderPaymentMethods = linkedProviderCustomer?.providerPaymentMethods || []
```

Comment:
Should we switch to early return like 

```typescript
if (!linkedProviderCustomer) return false

const availableProviderPaymentMethods = linkedProviderCustomer.providerPaymentMethods
```

I know we have a check after with linkedProvidercustomer but I feel like it would help just to stop thinking about this